source/shared/cpp/ObjectModel/FactSet.cpp

```diff
@@ -45,7 +45,14 @@ std::shared_ptr<BaseCardElement> FactSetParser::Deserialize(ParseContext& contex
 
     // Parse Facts
     auto facts =
-        ParseUtil::GetElementCollectionOfSingleType<Fact>(context, value, AdaptiveCardSchemaKey::Facts, Fact::Deserialize, true);
+        ParseUtil::GetElementCollectionOfSingleType<Fact>(context, value, AdaptiveCardSchemaKey::Facts, Fact::Deserialize, false);
+
+    if (facts.empty())
+    { 
+        context.warnings.emplace_back(std::make_shared<AdaptiveCardParseWarning>(WarningStatusCode::RequiredPropertyMissing,
+                                                                                  "required property, \"fact\", is missing"));
+    }
```

Comment:
Similar to my comment in #4765, i think we want to change the behavior of the ParseUtil function rather than change each individual caller. 

That said, the change in Fact.cpp would be an exception as in that case we do need specific additional logic to ensure that at least one of the two is set.

Reply:
About the exception, it will be still parse warning, and renderer will drop the rendering of this FactSet.

---
In reply to: [494601157](https://github.com/microsoft/AdaptiveCards/pull/4809#discussion_r494601157) [](ancestors = 494601157)

Reply:
I created a new issue that tracks array parsing similary to the PR4765(TextBlock). [https://github.com/microsoft/AdaptiveCards/issues/4837](https://github.com/microsoft/AdaptiveCards/issues/4837) 

---
In reply to: [496175681](https://github.com/microsoft/AdaptiveCards/pull/4809#discussion_r496175681) [](ancestors = 496175681,494601157)